core/src/main.cpp

```diff
@@ -1100,8 +1103,10 @@ auto main(int _argc, char* _argv[]) -> int
 		io_threads.join();
 
 		logging::trace("Cleaning up user mapper...");
-		auto close_mapper{irods::http::globals::user_mapping_lib().get<void ()>("user_mapper_close")};
-		close_mapper();
+		auto close_mapper{irods::http::globals::user_mapping_lib().get<int ()>("user_mapper_close")};
+		if (auto rc{close_mapper()}; rc != 0) {
+			logging::error("An error occured while closing the mapping plugin with code [{}]", rc);
```

Comment:
Should we return something non-zero here?

Reply:
Can do.